www/src/docs/common-pitfalls.md

```diff
@@ -0,0 +1,204 @@
+
+### Resources that are passthrough-copied
+
+A typical 11ty site has many static resources, including fonts, images, and more. Some of these (like images) may need to be processed manually using 11ty, but others (like fonts) just need to be copied to the build output directory.
+
+In Slinkity, there are two approaches you can take to ensure that static resources correctly make their way to the final build output:
+
+1. Using relative paths or import aliases to the resources on a page.
+2. Passthrough-copying the resources to a `/public` folder.
+
+#### Approach 1: Path resolution
+
+In the first approach, we lean on Vite's ability to resolve dependencies and resources referenced by our templates. In this case, if you want your font files to end up in your site's final build output, all you need to do is reference those font files somewhere in your CSS. You can do this using either relative paths (not absolute!) or [import aliases](/docs/import-aliases).
+
+For example, suppose you have a project structure that looks like this:
+
+```plaintext
+├── fonts
+│   └── Atkinson-Hyperlegible-Regular.woff2
+├── src
+│   ├── _includes
+│   │   └── layout.njk
+│   └── index.md
+└── styles
+    └── index.css
+```
+
+In the path resolution approach, you can either use relative paths:
+
+```css
+/* styles/index.css */
+@font-face {
+  font-family: Atkinson;
+  src: url('../fonts/Atkinson-Hyperlegible-Regular.woff2');
+  font-display: swap;
+}
+```
+
+Or you can leverage import aliases:
+
+```css
+/* styles/index.css */
+@font-face {
+  font-family: Atkinson;
+  src: url('/@root/fonts/Atkinson-Hyperlegible-Regular.woff2');
+  font-display: swap;
+}
+```
+
+As long as this stylesheet is later referenced somewhere in your layouts, Vite will handle the rest for you:
+
+{% raw %}
+```html
+<!-- src/_includes/layout.njk -->
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <!-- other tags omitted for brevity -->
+    <link rel="stylesheet" href="/@root/styles/index.scss">
+  </head>
+  <body>{{ content | safe }}</body>
+</html>
+```
+{% endraw %}
+
+In the production build, Vite will generate an `assets` directory containing your font files and CSS, nested in your build output directory:
+
+```plaintext
+_site
+├── assets
+│   ├── Atkinson-Hyperlegible-Regular.efe52c57.woff2
+│   └── index.7edd6dae.css
+└── index.html
+```
+
+The production CSS will then correctly reference the font files using root-relative paths, along with unique hashes that Vite appends to asset file names:
+
+```css
+/* _site/assets/index.7edd6dae.css */
+@font-face {
+  font-family: Atkinson;
+  src: url(/assets/Atkinson-Hyperlegible-Regular.efe52c57.woff2);
+  font-display: swap;
+}
+```
+
+> ⚠️ One downside to this approach: Vite can't resolve relative paths or import aliases from `preload` tags. If you need these on your site, we recommend going with the passthrough-copy approach instead.
+
+#### Approach 2: Passthrough-copying a `public` folder
```

Comment:
On closer inspection, I realized... this section is wrong! There's no need to passthrough copy to a `public` folder specifically, as long as your absolute paths reference whatever you're passthrough copying. 

Take these fonts for instance. Your `@font-face src` can happily point to `/fonts/Atkinson-Hyperlegible-Regular.woff2` as long as your passthrough copying to `fonts/`. Vite will discover that your CSS is referencing the file and copy this to the output. Right now, `robots.txt` and `sitemap.xml` are the only examples I can think of that would require a passthrough copy on `public/`. I think I'll rework this section to accomodate (but preserve your lovely folder trees 🙃

Reply:
Huh, you're right! Just tested in a sandbox. Apparently Vite _is_ clever enough to pick up on that relationship. It even updates the font-face CSS paths to correctly reference `/assets/whatever.woff2`.